keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,604 @@
+- Change the existing behavior of `Indexed` mode without a SuccessPolicy.
+
+## Proposal
+
+We propose new policies under which a job can be declared as succeeded.
+Those policies can be modeled in the following:
+
+1. An indexed job completes if a set of [x, y, z...] indexes are successful.
+2. An indexed job completes if x or x% of indexes are successful.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a machine-learning researcher, I run an indexed job which drivers (leaders) are launched
+as index=0,1,2, and workers are launched as index=3+.
+I want to care about only the drivers when the result of job is evaluated.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    onSucceededPods:
+      indexes: "0-2"
+      whenTargetAchieved: terminate
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./sample"]
+```
+
+#### Story 2
+
+As a machine-learning researcher, I run an indexed job with an autoscale semantic
+set as a minimum number is 5 and a maximum number is 20.
+I want to mark the job as successful at least 5 indexes succeeded, even if 
+all indexes don't succeed.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    onSucceededPods:
+      count: 5
+      whenTargetAchieved: terminate
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: job-container
+        image: job-image
+        command: ["./sample"]
+```
+
+#### Story 3
+
+As a machine-learning researcher from the [Story 2](#story-2), I want to mark the job
+as successful if 50% of all indexes succeed because I can not know how many indexes 
+will have in advance.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    onSucceededPods:
+      count: 50%
+      whenTargetAchieved: terminate
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: job-container
+          image: job-image
+          command: ["./sample"]
+```
+
+#### Story 4
+
+As a simulation researcher/engineer for fluid dynamics/biochemistry.
+I want to mark the job as successful and continue running the rest of the pods if some indexes succeed
+because I set the minimum required value for sampling in the `.successPolicy.onSucceededPods.count` and
+perform the same simulation in all indexes.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  parallelism: 10
+  completions: 10
+  completionMode: Indexed
+  successPolicy:
+    onSucceededPods:
+      count: 5
+      whenTargetAchieved: continue
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: job-container
+          image: job-image
+          command: ["./sample"]
+```
+
+### Notes/Constraints/Caveats (Optional)
```

Comment:
Add a note that the concepts `completed` or `succeeded` job are used in the KEP interchangeably. 

It might be surprising to a reader that we change between these two terms. Even more, we do so at the API level calling the new policy `SuccessPolicy`, while the condition is called Completed, we also have `CompletedIndexes` field.

Reply:
Done.